hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileDataBlock.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.io.hfile;
+
+import org.apache.hudi.io.util.IOUtils;
+
+import java.util.Optional;
+
+import static org.apache.hudi.io.hfile.KeyValue.KEY_OFFSET;
+
+/**
+ * Represents a {@link HFileBlockType#DATA} block in the "Scanned block" section.
+ */
+public class HFileDataBlock extends HFileBlock {
+  protected HFileDataBlock(HFileContext context,
+                           byte[] byteBuff,
+                           int startOffsetInBuff) {
+    super(context, HFileBlockType.DATA, byteBuff, startOffsetInBuff);
+  }
+
+  /**
+   * Seeks to the key to look up.
+   *
+   * @param key Key to look up.
+   * @return The {@link KeyValue} instance in the block that contains the exact same key as the
+   * lookup key; or empty {@link Optional} if the lookup key does not exist.
+   */
+  public Optional<KeyValue> seekTo(Key key) {
+    int offset = startOffsetInBuff + HFILEBLOCK_HEADER_SIZE;
+    int endOffset = offset + onDiskSizeWithoutHeader;
+    // TODO: check last 4 bytes in the data block
+    while (offset + HFILEBLOCK_HEADER_SIZE < endOffset) {
+      // Full length is not known yet until parsing
+      KeyValue kv = new KeyValue(byteBuff, offset, -1);
+      // TODO: Reading long instead of two integers per HBase
+      int comp =
+          IOUtils.compareTo(kv.getBytes(), kv.getKeyContentOffset(), kv.getKeyContentLength(),
+              key.getBytes(), key.getContentOffset(), key.getContentLength());
+      if (comp == 0) {
+        return Optional.of(kv);
+      } else if (comp > 0) {
+        return Optional.empty();
+      }
+      // TODO: check what's the extra byte
```

Comment:
so, you need to read an extra byte. always? is it some kind of type/enum? Would be good to understand what this is.

Reply:
It's the HFile MVCC version (timestamp represented in a variable-length number).  Given we don't use it, it's always 0, thus the length is always 1.